app/models/instance.rb

```diff
@@ -15,6 +15,10 @@ def find_tenant_by_host(host)
     end
   end
 
+  VALID_HOST_NAME_REGEX = /\A[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}\z/i
+  validates :host, presence: true, format: { with: VALID_HOST_NAME_REGEX },
+    uniqueness: { case_sensitive: false }
```

Comment:
Set a unique case sensitive index on the database table. no need to validate presence and uniqueness here, because the schema_validations gem does it.

the uniqueness validation is not thread safe anyway.